llvm/test/MC/AMDGPU/gfx9-asm-err.s

```diff
@@ -41,3 +41,6 @@ global_load_dword v[2:3], off
 
 scratch_load_dword v2, off, offset:256
 // GFX9ERR: :[[@LINE-1]]:{{[0-9]+}}: error: too few operands for instruction
+
+s_sendmsg sendmsg(MSG_SYSMSG, SYSMSG_OP_HOST_TRAP_ACK)
+// GFX9ERR: :[[@LINE-1]]:{{[0-9]+}}: error: invalid operation id
```

Comment:
Error message leaves something to be desired. Can it mention the lack of support on the target, that shows it is recognized?

Reply:
Sure, fixed in the update. I ended up doing a bit of refactoring to support this cleanly.